wsagent/che-core-api-project/src/test/java/org/eclipse/che/api/project/server/ProjectServiceTest.java

```diff
@@ -928,6 +939,50 @@ public void testDeleteProject() throws Exception {
     }
 
     @Test
+    public void testDeleteProjectsConcurrently() throws Exception {
+        int threadNumber = 100;
+        ExecutorService executor = Executors.newFixedThreadPool(threadNumber);
+        CountDownLatch countDownLatch = new CountDownLatch(threadNumber);
+        List<Future<ContainerResponse>> futures = new LinkedList<>();
+
+
+        for (int i = 0; i < threadNumber; i++) {
+            addMockedProjectConfigDto(ptRegistry.getProjectType("my_project_type"), "my_project_name" + i);
+        }
+
+        IntStream.range(0, threadNumber).forEach(
```

Comment:
my concern was only to unify the two loops into a single loop. (because we iterate on the same numbers but in a different way)
and yes to avoid hacks like int j = i; you've to use IntStream